File: benches/fixtures/sample.py
import os
import subprocess
import html

def get_env_value():
    return os.environ.get("SECRET_KEY", "")

def sanitize_input(val):
    return html.escape(val)

def execute_command(cmd):
    subprocess.run(cmd, shell=True)

def safe_flow():
    val = get_env_value()
    clean = sanitize_input(val)
    print(clean)

def unsafe_flow():
    val = get_env_value()
    execute_command(val)

if __name__ == "__main__":
    safe_flow()
    unsafe_flow()

File: benches/fixtures/sample.java
import java.io.IOException;

public class Sample {
    public static String getEnv() {
        return System.getenv("DB_PASSWORD");
    }

    public static String sanitize(String input) {
        return input.replaceAll("[<>&]", "");
    }

    public static void executeCommand(String cmd) throws IOException {
        Runtime.getRuntime().exec(cmd);
    }

    public static void safeFlow() throws IOException {
        String val = getEnv();
        String clean = sanitize(val);
        System.out.println(clean);
    }

    public static void unsafeFlow() throws IOException {
        String val = getEnv();
        executeCommand(val);
    }

    public static void main(String[] args) throws IOException {
        safeFlow();
        unsafeFlow();
    }
}
